### ISSUE TITLE
🌟 Powered by CKEditor

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#14082]
## tl;dr

From version v38.0.0 onwards, all open-source installations of CKEditor (using it under the GPL terms) will carry a small “Powered by CKEditor” logo in the bottom right corner of the editing area. This new logo links directly to the [CKEditor website](https://ckeditor.com/) to strengthen the awareness of who’s powering and modernizing their rich text editor.

This change does not affect customers with commercial licenses. You can reach out to our Technical Support team, using [this form](https://ckeditor.com/contact/), if you have any questions.

## Why does the branding matter to the project?

As a component, CKEditor is always part of bigger systems – applications like CMSs, CRMs, DMSs, blogs, chat apps, and mail clients. While the brand names of those end-product systems are well known, only a small percentage of users know they’re also CKEditor users.

We have discovered that this holds true even when CKEditor is an essential component of those systems. I can recollect numerous eye-opening conversations with people who turned out to be CKEditor users at conferences, meetups, or simply during conversations with our potential clients.

We’re absolutely committed to keeping CKEditor free and open source and plan to maintain it for many years to come. But all that comes at a cost. Currently, we hire well over 20 full-time employees who contribute to CKEditor every day. We’re also 8 years into the development of the project (and 20 years if you count all CKEditor versions).

Over the years, we designed a business model that allowed us to maintain our amazing team while offering CKEditor 5 to the world under the open-source GPL license. However, two important ingredients of this model are: the brand awareness and visibility of the commercial offer.

CKEditor 5 enables developers to create tight and highly customized rich text editing solutions within their apps. Again, it’s a component of your systems and we take that into consideration every time we design a new feature. However, that flexibility, customizability, and nature of the project play against our need to maintain the awareness of the CKEditor brand. A deep integration of an editor creates amazing UX but makes it blend completely into its surroundings.

## The solution

In the past, CKEditor 4 came with an “About” button in the toolbar. We don’t want to take that precious space and therefore we looked for a different solution. Again, the flexibility and number of “flavors” of CKEditor 5 became a problem.

We looked for an option that would work in as many cases as possible, that’s least intrusive, and ideally, within the realms of CKEditor 5 itself (within its UI). We decided to go with a small watermark-like logo, only visible for a focused editor, placed in the bottom right corner of the editing area.

Again, we’re aware that CKEditor needs to integrate well with its surroundings and thus, we want to make the placement and styling of the banner configurable. We’ll provide a guide explaining the possible options.

We’re aware that this is a significant change for some projects and we’re open to feedback too.

## How it looks

The placement of the logo is configurable, but by default, it’ll look like this:

![CKEditor logo in the bottom right corner (inside) the editable area](https://github.com/ckeditor/ckeditor5/assets/156149/01b336cc-5150-4e56-9d49-34b3f39ad3b9)

Or like this:

![CKEditor logo over the border of the editable area](https://github.com/ckeditor/ckeditor5/assets/156149/9fec9d85-70f7-4d30-b04f-ae204a16dbe2)

We’re still reviewing the options and testing them with different setups, addons (e.g. Grammarly, WebSpellChecker, LanguageTool), and editor configurations, so the final solution may change a bit.

### I don’t want the logo on my editor. What should I do?

If you’d prefer not to have the logo, [contact our team](https://ckeditor.com/contact-sales/#contact-form) to discuss your options.

### I have a commercial license for CKEditor. How do I remove the logo?

1.  You need to configure the `config.licenseKey`.
2.  You will find the new license key in the [CKEditor 5 Ecosystem dashboard](https://dashboard.ckeditor.com/login).

**Note:** You only need to do this if/when you are updating to v38.0.0+.

### This logo concerns me. Who/how can I give feedback on it?

We’re aware this is a significant change for some projects and we’re open to feedback. Feel free to comment on this issue.

**Comment by zadam** [comment]
I'll be honest with you and say that I don't like it. 

I remember ~ 6 years ago when I was choosing the editing component for my OSS project, the big attraction was that CKEditor provided a clean and distraction-free experience with its BalloonEditor. That benefit will be downgraded now. 

I understand that you need to be credited for your hard work and contribution to open source (my project's github contains a [shout-out section](https://github.com/zadam/trilium#-shoutouts) linking CKEditor). 

I guess my issue with this is that this is displayed non-stop. I would prefer a solution where the credit is displayed in the balloon/block toolbar, so that it is displayed only once the user interacts with these editing features.

**Comment by Reinmar** [comment]
Thanks for the honest feedback, @zadam. And your contributions over the years because you're one of the most active members of our community.

The decision didn't come lightly. We're well aware of how it affects the integrations and worked a lot on trying to balance its impact and effectiveness.

I'd like to clarify two things that I didn't write much about in the annoucement:

*   The logo appears only for a focused editor. It's hidden when a user does not interact with the editor.
*   The logo is not shown for small editors (below 50px x 250px AFAIR).

In both cases, our intention was to avoid displaying it in scenarios where it could be too distractive.

Also, we intentionally will keep it styleable so it can be incorporated into a layout in scenarios where the editor is heavily customized.

> I would prefer a solution where the credit is displayed in the balloon/block toolbar, so that it is displayed only when the user interacts with these editing features.

We were considering positioning it somewhere next to a toolbar but couldn't find a reasonable design for it. But I don't see why we couldn't have this as an option. Do you have an idea how it could look like?

**Comment by zadam** [comment]
@Reinmar I appreciate your answer.

What I had in mind was something like this:

![image](https://github.com/ckeditor/ckeditor5/assets/617641/36b03bb8-8236-4212-991c-7c128ab15430)

This way, the logo is displayed only when interacting with advanced / rich features. The same could be used in Block toolbar. Not sure if it makes sense from the design / UX perspective. 

As a benefit, this makes the logo a bit more prominent - with it being displayed right in the center where the user is looking, not just looming in the vision periphery in the corner.

For the ClassicEditor, something like this:

![image](https://github.com/ckeditor/ckeditor5/assets/617641/6c821382-83f2-45e7-a26a-1b48c5d2bda4)

Basically the credit steals a bit of the width and the toolbar would be forced to wrap around with more items.

Just my 2 cents. Not sure if other users share my views ...

**Comment by Reinmar** [comment]
> What I had in mind was something like this:
> 
> ![image](https://user-images.githubusercontent.com/617641/237818991-36b03bb8-8236-4212-991c-7c128ab15430.png)
> 
> This way, the logo is displayed only when interacting with advanced / rich features. The same could be used in Block toolbar. Not sure if it makes sense from the design / UX perspective, though.

TBH, this would not be acceptable to me as a user because the logo obstructs the clickable area of buttons in the toolbar. We discussed several ideas related to placing the logo next to the toolbar (further from it than on your mockup) but the team considered them too intrusive – the logo would get in a way of a person working with the toolbar. And this is the area where you potentially click the most.

So, the only option I'd consider at the moment is this one:

> For the ClassicEditor, something like this:
> 
> ![image](https://user-images.githubusercontent.com/617641/237820491-6c821382-83f2-45e7-a26a-1b48c5d2bda4.png)
> 
> Basically the credit steals a bit of the width and the toolbar would be forced to wrap around with more items.

Though, it has an application only in case where the editor is wide and does not contain too many buttons. And from my experience, this is a relatively small fraction of cases. That's why we were looking for a more universal solution.

---

As for Trillium, I asked guys on the team to play a bit with it and with these styles:

```
.ck.ck-balloon-panel.ck-powered-by-balloon {
    --ck-powered-by-background: var(--accented-background-color);
}

.ck.ck-balloon-panel.ck-powered-by-balloon .ck.ck-powered-by a::after {
    display: none;
}

.ck.ck-editor__editable.note-detail-editable-text-editor {
    padding-bottom: 10px;
}

.ck.ck-balloon-panel.ck-powered-by-balloon .ck.ck-powered-by .ck-powered-by__label {
    text-transform: none; 
    font-family: var(--main-font-family);   
    padding-left: 2px;
    color: var(--main-text-color);
}

.ck.ck-balloon-panel.ck-powered-by-balloon .ck.ck-powered-by .ck-icon {
    filter: brightness(10);
}
```

They achieved this:

![image](https://github.com/ckeditor/ckeditor5/assets/156149/deac1955-d5e5-44ff-a078-483d93de661b)

Which in our opinion looks good. It's doesn't stand out too much and it's not in a way of user editing. I hope it helps.

**Comment by mlewand** [comment]
We had a call today regarding design, here are conclusions:

*   We go back to over-the-border position by default.
*   We'll open an issue in official repository for possible improvment: positioning PBC logo at the top of the document (between toolbar and content).
*   Configurable "Powered by" label text.

**Comment by tomatkapitalise** [comment]
Hi, thanks for the open feedback opportunity.

Looking at this I understand the need from your PoV, but I have a few issues. First and foremost I believe it is a mistake to have the text appear when the input is focused. A small piece of text appearing in the corner every time you click a textbox really threw the colleagues I showed this to.

Next, the balloon leaks outside the container. Again, my opinion, but components should occupy their defined space and not have text overlapping other elements.

The first image you showed seemed best, if there was a way to maybe select this as an option that'd be great.

**Comment by Reinmar** [comment]
Thank you for your honest feedback @winzing. We truly appreciate it.

Pricing is a complex and crucial business decision and we understand that our current pricing model may not cater to every use case. We have been continuously working on improving it based on valuable feedback like yours and we're committed to meeting the needs of our users.

At the same time, we highly value open-source principles and thus, we do offer the core functionality of our product in line with the "freedom of software" for everyone that complies with GPL v2+. In return, we kindly request that the use of CKEditor is visibly attributed. This is a common practice in the market and helps us ensure the continued development and support of our product.

You mentioned that you use CKEditor on your forums. I'd like to better understand how cases similar to yours can be affected by "Powered by CKEditor" logo. Could you please connect with me via p.koszulinski@cksource.com?

**Comment by winzig** [comment]
Sure I will reach out

**Comment by quicksketch** [comment]
Voicing my opinion as a CMS maintainer and previously involved in bringing CKEditor to Drupal. I think CKEditor promoting their brand by default on new instances is 100% okay, and if it encourages the growth and support of the CKEditor project we should support this decision by CKSource.

Most software includes a public promotion of the software out-of-the-box. In the case of open-source software, there is almost always a simple way to disable this promotion. Under the GPL license, re-distributors have the right to make any modifications they want, so attempting to force attribution doesn't usually work. So like other projects, CKEditor can and absolutely should self-promote, but at the same time, users wanting to disable the attribution should be able to do so easily. The current implementation definitely qualifies as easy to disable. The current badge is as easy to hide as simple as this:

```
.ck-powered-by {
  display: none;
}
```

Whether to display the badge seems like it's going to be a decision for each project that uses CKEditor. But as long as it remains a choice, there's nothing to complain about.

**Comment by winzig** [comment]
At this point the bigger issue for me is not the attribution but the licensing. https://github.com/ckeditor/ckeditor5/issues/14314

**Comment by kesselb** [comment]
Could you add `rel=noreferrer` to the powered by link to omit the referrer?
